src/units/abilities.cpp

```diff
@@ -705,15 +705,15 @@ std::string attack_type::weapon_specials(bool only_active, bool is_backstab) con
  */
 attack_type::specials_context_t::specials_context_t(const attack_type& weapon,
                                        const_attack_ptr other_attack,
-									   const unit& self,
-                                       const unit& other,
+									   const unit* self,
```

Comment:
Your other PR is trying to use `unit_ptr` everywhere, yet here you're using `unit*`?